nightfall-client/src/services/burn.mjs

```diff
@@ -0,0 +1,138 @@
+import axios from 'axios';
+import config from 'config';
+import constants from '@polygon-nightfall/common-files/constants/index.mjs';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import { waitForContract } from '@polygon-nightfall/common-files/utils/contract.mjs';
+import gen from 'general-number';
+import Transaction from '@polygon-nightfall/common-files/classes/transaction.mjs';
+import { clearPending } from './commitment-storage.mjs';
+import { getCommitmentInfo } from '../utils/getCommitmentInfo.mjs';
+import { computeCircuitInputs } from '../utils/computeCircuitInputs.mjs';
+import { submitTransaction } from '../utils/submitTransaction.mjs';
+
+const { ZOKRATES_WORKER_HOST, PROVING_SCHEME, BACKEND, PROTOCOL, VK_IDS } = config;
+const { SHIELD_CONTRACT_NAME } = constants;
+const { generalise } = gen;
+
+async function burn(burnParams) {
+  logger.info('Creating a burn transaction');
+  // let's extract the input items
+  const { providedCommitments, ...items } = burnParams;
+  const { rootKey, value, fee, ercAddress, tokenId } = generalise(items);
+
+  // now we can compute a Witness so that we can generate the proof
+  const shieldContractInstance = await waitForContract(SHIELD_CONTRACT_NAME);
+
+  const maticAddress = generalise(
+    (await shieldContractInstance.methods.getMaticAddress().call()).toLowerCase(),
+  );
+
+  const commitmentsInfo = await getCommitmentInfo({
+    totalValueToSend: generalise(value).bigInt,
+    fee,
+    ercAddress,
+    tokenId,
+    maticAddress,
+    rootKey,
+    maxNumberNullifiers: VK_IDS.burn.numberNullifiers,
+    onlyFee: true,
+    providedCommitments,
+  });
+
+  // Burn will have two commitments. The first will belong to the change if commitment isn't fully burnt, and the second one to the fee.
```

Comment:
What does `commitment not fully burnt` mean?

Reply:
L2 mintings will be work in a similar way as ERC1155, meaning you can have multiple "l2 assets" belonging to the same l2 address and tokenId. So, when you burn a commitment, you might not want to burn the full amount